tests/common/request.py

```diff
@@ -75,16 +78,19 @@ def send_transaction(
     call_data = (
         None
         if method_name is None and method_args is None
-        else [method_name, json.dumps(method_args)]
+        else [calldata.encode({"method": method_name, "args": method_args})]
     )
     signed_transaction = sign_transaction(account, call_data, contract_address, value)
     return send_raw_transaction(signed_transaction)
 
 
 def deploy_intelligent_contract(
-    account: Account, contract_code: str, constructor_params: str
+    account: Account, contract_code: str, method_args: list
 ) -> tuple[str, dict]:
-    deploy_data = [contract_code, constructor_params]
+    deploy_data = [
+        contract_code,
+        calldata.encode({"method": "__init__", "args": method_args}),
```

Comment:
For now I would prefer to stick to hardcoded `__init__` for a constructor. I am thinking about changing some abi properties of calldata, so we may change it later